lib/chef/provider/script.rb

```diff
@@ -66,10 +67,45 @@ def action_run
       end
 
       def set_owner_and_group
-        # FileUtils itself implements a no-op if +user+ or +group+ are nil
-        # You can prove this by running FileUtils.chown(nil,nil,'/tmp/file')
-        # as an unprivileged user.
-        FileUtils.chown(new_resource.user, new_resource.group, script_file.path)
+        if ! Chef::Platform.windows?
```

Comment:
Maybe this could be read better like:
```
if Chef::Platform.windows?
  grant_alternate_user_read_access
else
  FileUtils.chown(new_resource.user, new_resource.group, script_file.path)
end
```